### PR TITLE
prov/efa: Two bugfixes in local read logic

### PR DESCRIPTION
**Pull request by zhngaj**
9d04a6c31f1d3e0efd8f00ba0df4d818373bc7c0 prov/efa: Fix a bug of using uninitialized field of rxr_read_entry
dde4cedad6290fefc570cd3129075acf7a519b23 prov/efa: Fix a bug in rxr_handle_rma_read_completion

### CODE REVIEW

**Review by wzamazon — APPROVED** [review]

**Comment by zhngaj** [comment]
@shefty pls let me know if the CI failure in continuous-integration/appveyor/pr is a real failure caused by this PR?

File: prov/efa/src/rxr/rxr_read.c [hunk 1]
```diff
@@ -380,6 +380,7 @@ int rxr_read_post_local_read_or_queue(struct rxr_ep *ep,
 		FI_WARN(&rxr_prov, FI_LOG_EP_CTRL, "RDMA entries exhausted\n");
 		return -FI_ENOBUFS;
 	}
+	memset(read_entry, 0, sizeof(struct rxr_read_entry));
```
Comment (bwbarrett) [inline prov/efa/src/rxr/rxr_read.c:383]: This is a fairly big structure (320 bytes on x86_64 linux), we fill in most of the fields below, and the remaining fields are not necessary to fill in.  I'm not sure this is the right solution.  If you want to (during memory poisoning) memset to some weird large number to catch these issues, I'm ok with that.  But this does not feel like the right solution for critical path code.

**Review by bwbarrett — CHANGES_REQUESTED** [review]

File: prov/efa/src/rxr/rxr_read.c (hunk 1 above)
Comment (zhngaj) [reply prov/efa/src/rxr/rxr_read.c]: Fixed by only memsetting mr field, which was the problem causing field.

**Review by bwbarrett — APPROVED** [review]

**Comment by shefty** [comment]
If all changes are contained within prov/efa, you can assume that any Intel CI failure is unrelated.  The only exception I can think of is possible failures with make distcheck, and if other CI covers that case as well, we should be okay.  (I should be able to add a github action to test that.)

**Comment by zhngaj** [comment]
> If all changes are contained within prov/efa, you can assume that any Intel CI failure is unrelated. The only exception I can think of is possible failures with make distcheck, and if other CI covers that case as well, we should be okay. (I should be able to add a github action to test that.)

Thanks, the changes are only within prov/efa.

**Pull request merged**